projects/storefrontlib/src/cms-components/checkout/components/checkout-progress/checkout-progress.component.ts

```diff
@@ -1,43 +1,48 @@
-import { RoutingService, RoutingConfigService } from '@spartacus/core';
-import { Component, ChangeDetectionStrategy, OnInit } from '@angular/core';
-import { CheckoutConfig } from '../../config/checkout-config';
+import {
+  Component,
+  ChangeDetectionStrategy,
+  OnInit,
+  ChangeDetectorRef,
+  OnDestroy,
+} from '@angular/core';
+import { CheckoutStepService } from '../../services/checkout-step.service';
 import { CheckoutStep } from '../../model/checkout-step.model';
-import { Observable } from 'rxjs';
+import { Observable, Subscription } from 'rxjs';
 import { tap } from 'rxjs/operators';
 
 @Component({
   selector: 'cx-checkout-progress',
   templateUrl: './checkout-progress.component.html',
   changeDetection: ChangeDetectionStrategy.OnPush,
 })
-export class CheckoutProgressComponent implements OnInit {
+export class CheckoutProgressComponent implements OnInit, OnDestroy {
   constructor(
-    protected config: CheckoutConfig,
-    protected routingService: RoutingService,
-    protected routingConfigService: RoutingConfigService
```

Comment:
These are part of the breaking changes that will be addressed in the follow-up ticket https://github.com/SAP/spartacus/issues/7509 , right?

Reply:
I will leave these breaking changes later, because this "checkout-progress" stuff will be refactored in #4895